cmd/earthly/build_cmd.go

```diff
@@ -225,14 +225,14 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 	app.console.PrintPhaseHeader(builder.PhaseInit, false, "")
 	app.warnIfArgContainsBuildArg(flagArgs)
 
-	err = app.initFrontend(cliCtx)
+	err = app.configureSatellite(cliCtx, cloudClient, gitCommitAuthor, gitConfigEmail)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "could not construct new buildkit client")
 	}
 
-	err = app.configureSatellite(cliCtx, cloudClient, gitCommitAuthor, gitConfigEmail)
+	err = app.initFrontend(cliCtx)
 	if err != nil {
-		return errors.Wrapf(err, "could not construct new buildkit client")
+		return err
```

Comment:
nit: I like to see wrapped errors to provide extra context about what failed. In this case, it may be as small as `errors.Wrap(err, "could not initialize frontend")`, but it still helps a little when tracking down the source of an error.

Reply:
I agree, this was just part of moving the order of the calls and their matching error handling if clauses. I can add a wrapping around this error